sdk/keyvault/azure-keyvault-keys/src/samples/java/com/azure/security/keyvault/keys/KeyClientJavaDocCodeSnippets.java

```diff
@@ -89,9 +100,337 @@ public KeyAsyncClient createAsyncClientWithPipeline() {
     public void createKey() {
         KeyClient keyClient = createClient();
         // BEGIN: com.azure.keyvault.keys.keyclient.createKey#string-keyType
-        Key retKey = keyClient.createKey("keyName", KeyType.EC).value();
-        System.out.printf("Key is created with name %s and id %s %n", retKey.name(), retKey.id());
+        Key key = keyClient.createKey("keyName", KeyType.EC);
+        System.out.printf("Key is created with name %s and id %s %n", key.name(), key.id());
         // END: com.azure.keyvault.keys.keyclient.createKey#string-keyType
+
+        // BEGIN: com.azure.keyvault.keys.keyclient.createKey#string-keyOptions
+        KeyCreateOptions keyCreateOptions = new KeyCreateOptions("keyName", KeyType.RSA)
+            .notBefore(OffsetDateTime.now().plusDays(1))
+            .expires(OffsetDateTime.now().plusYears(1));
+        Key optionsKey = keyClient.createKey(keyCreateOptions);
+        System.out.printf("Key is created with name %s and id %s \n", optionsKey.name(), optionsKey.id());
+        // END: com.azure.keyvault.keys.keyclient.createKey#string-keyOptions
+
+        // BEGIN: com.azure.keyvault.keys.keyclient.createRsaKey#keyOptions
+        RsaKeyCreateOptions rsaKeyCreateOptions = new RsaKeyCreateOptions("keyName")
+            .keySize(2048)
+            .notBefore(OffsetDateTime.now().plusDays(1))
+            .expires(OffsetDateTime.now().plusYears(1));
+        Key rsaKey = keyClient.createRsaKey(rsaKeyCreateOptions);
+        System.out.printf("Key is created with name %s and id %s \n", rsaKey.name(), rsaKey.id());
+        // END: com.azure.keyvault.keys.keyclient.createRsaKey#keyOptions
+
+        // BEGIN: com.azure.keyvault.keys.keyclient.createEcKey#keyOptions
+        EcKeyCreateOptions ecKeyCreateOptions = new EcKeyCreateOptions("keyName")
+            .curve(KeyCurveName.P_384)
+            .notBefore(OffsetDateTime.now().plusDays(1))
+            .expires(OffsetDateTime.now().plusYears(1));
+        Key ecKey = keyClient.createEcKey(ecKeyCreateOptions);
+        System.out.printf("Key is created with name %s and id %s \n", ecKey.name(), ecKey.id());
+        // END: com.azure.keyvault.keys.keyclient.createEcKey#keyOptions
+
+        // BEGIN: com.azure.keyvault.keys.keyclient.importKey#keyImportOptions
+        // JsonWebKey jsonWebKeyToImport;
```

Comment:
These code snippets would need a valid `jsonWebKeyToImport`  to work.
@g2vinay do we have a good working example for this or should we remove this for now?

Reply:
We will have a good working example with Crypto API implementation.
Where we can import local crypto keys into the service. I'll let you know, once I add that.
For now, let's keep it in the code files. Add a TODO there.